src/cache/diagnostic_edmf_precomputed_quantities.jl

```diff
@@ -314,31 +325,52 @@ function set_diagnostic_edmf_precomputed_quantities!(Y, p, turbconv_model)
                 )
 
             @. u³ʲ_datau³ʲ_data +=
-                (1 / local_geometry_halflevel.J^2) * (
-                    local_geometry_prev_level.J^2 *
+                (
+                    1 /
+                    (local_geometry_halflevel.J * local_geometry_halflevel.J)
+                ) * (
+                    local_geometry_prev_level.J *
+                    local_geometry_prev_level.J *
                     FT(2) *
                     (
                         entr_detrʲ_prev_level.entr * u³⁰_data_prev_halflevel -
                         entr_detrʲ_prev_level.entr * u³ʲ_data_prev_halflevel
                     )
                 )
-
-            @. u³ʲ_halflevel = ifelse(
-                (
-                    u³ʲ_datau³ʲ_data < (FT(1e-6) / ∂x³∂ξ³_level^2) ||
+            # Helper function for broadcasting ternary operator across fields.
+            # This is necessary to prevent allocations using ifelse().
+            function pointwise_ifelse(
```

Comment:
Is this needed? I haven't seen `ifelse` allocate in the past

Reply:
Sorry, I don't remember seeing these changes when I approved the PR, this looks highly specialized and it's not clear to me that it's needed